lib/pure/collections/sequtils.nim

```diff
@@ -509,12 +509,37 @@ proc keepIf*[T](s: var seq[T], pred: proc(x: T): bool {.closure.})
       inc(pos)
   setLen(s, pos)
 
-func delete*[T](s: var seq[T]; first, last: Natural) =
+func delete*[T](s: var seq[T]; slice: Slice[int]) =
+  ## Deletes the items `s[slice]`, raising a defect if the slice contains
+  ## elements out of range.
+  runnableExamples:
+    var a = @[10, 11, 12, 13, 14]
+    doAssertRaises(AssertionDefect): a.delete(4..5)
+    assert a == @[10, 11, 12, 13, 14]
+    a.delete(4..4)
+    assert a == @[10, 11, 12, 13]
+    a.delete(1..2)
+    assert a == @[10, 13]
+    a.delete(1..<1) # empty slice
+    assert a == @[10, 13]
+  assert slice.a < s.len and slice.a >= 0 and slice.b < s.len, $(slice, s.len)
+  var i = slice.a
+  var j = min(len(s), slice.b + 1)
```

Comment:
```suggestion
  var j = slice.b + 1
```
We check for `slice.b < s.len`, so `slice.b + 1 <= len(s)` always holds.

Reply:
done; i was hesitant to avoid having bad behavior with -d:danger but you're right, that's the point of running tests with assertions on, and plenty of other things would break with violated asserts anyways.